kotlin-android/app/src/main/java/com/mongodb/tasktracker/ProjectActivity.kt

```diff
@@ -0,0 +1,114 @@
+            Log.w(TAG(), e)
+        }
+        if (user == null) {
+            // if no user is currently logged in, start the login activity so the user can authenticate
+            startActivity(Intent(this, LoginActivity::class.java))
+        } else {
+            // configure realm to use the current user and the partition corresponding to "My Project"
+            val config = SyncConfiguration.Builder(user!!, "user=${user!!.id}")
+                .waitForInitialRemoteData()
+                .build()
+
+            // save this configuration as the default for this entire app so other activities and threads can open their own realm instances
+            Realm.setDefaultConfiguration(config)
+
+            // Sync all realm changes via a new instance, and when that instance has been successfully created connect it to an on-screen list (a recycler view)
+            Realm.getInstanceAsync(config, object: Realm.Callback() {
+                override fun onSuccess(realm: Realm) {
+                    // since this realm should live exactly as long as this activity, assign the realm to a member variable
+                    this@ProjectActivity.realm = realm
+                    setUpRecyclerView(realm)
+                }
+            })
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_project)
+
+        recyclerView = findViewById(R.id.project_list)
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        recyclerView.adapter = null
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        menuInflater.inflate(R.menu.activity_task_menu, menu)
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            R.id.action_logout -> {
+                user?.logOutAsync {
+                    if (it.isSuccess) {
+                        user = null
+                        Log.v(TAG(), "user logged out")
+                        startActivity(Intent(this, LoginActivity::class.java))
+                    } else {
+                        Log.e(TAG(), "log out failed! Error: ${it.error}")
+                    }
+                }
+                true
+            }
+            else -> {
+                super.onOptionsItemSelected(item)
+            }
+        }
+    }
+
+    private fun setUpRecyclerView(realm: Realm) {
+        Log.e(TAG(), "User id: " + user?.id)
+        val syncedUser = realm.where<User>().sort("_id").findFirstAsync()
+
+        // hacky solution for findFirst not actually blocking until it returns a valid result
+        syncedUser.addChangeListener { syncedUserData : User ->
+            if (syncedUserData.isLoaded()) {
+                val projectsList = syncedUserData?.memberOf ?: RealmList<Project>(Project("My Project", "user=${user?.id}"))
```

Comment:
I suspect this is supposed to be 
```suggestion
                val projectsList = syncedUserData?.memberOf ?: RealmList<Project>(Project("My Project", "project=${user?.id}"))
```

Reply:
Project activity displays the projects owned by a user (and as such only needs to synchronize the realm containing the user's User custom user data object), so I *think* this is actually right. The TaskActivity needs project=, though.